bids/tests/test_config.py

```diff
@@ -66,7 +67,7 @@ def test_extension_initial_dot(mock_config):
         bids.config.set_option('extension_initial_dot', True)
 
     # No warnings on layout construction
-    with pytest.warns(None) as record:
+    with warnings.catch_warnings():
```

Comment:
according to https://docs.pytest.org/en/latest/how-to/capture-warnings.html#additional-use-cases-of-warnings-in-tests